### ISSUE TITLE
failed to import TS file from node_modules on Linux & Mac

### ISSUE THREAD

**Issue by TechQuery** [TypeStrong/ts-node#990]
## Expected Behavior

I want to import everything (including **TS source codes** in `node_modules/`) :

```javascript
register({
    ignore: [],
    compilerOptions: { module: 'CommonJS' }
});
```
https://github.com/EasyWebApp/MarkCell/blob/7b43ea0/source/File.ts#L26-L29

## Actual Behavior

| OS           | result                                                                                                     |
|:----------:|:---------------------------------------------------------------------------:|
| Windows | everything is OK                                                                                   |
| Linux       | `TypeError: Unable to require file: node_modules/xxx/source/yyy.tsx` |
| Mac        | `TypeError: Unable to require file: node_modules/xxx/source/yyy.tsx` |

### error on Linux

https://travis-ci.com/github/EasyWebApp/mark-wiki/builds/155135722

### error on Mac

![image](https://user-images.githubusercontent.com/19969570/77620795-19544a80-6f76-11ea-881e-2056ecd73b55.png)

## Steps to reproduce the problem

```shell
git clone https://github.com/EasyWebApp/mark-wiki.git
cd mark-wiki/
npm install
npm run build
```

## Minimal reproduction

https://github.com/TypeStrong/ts-node-repros/pull/1

## Specifications

* ts-node version: 8.8.1
* TypeScript version: 3.8.3
* tsconfig.json, if you're using one:
```json
{
    "compilerOptions": {
        "module": "ESNext",
        "moduleResolution": "Node",
        "resolveJsonModule": true,
        "experimentalDecorators": true,
        "jsx": "react",
        "jsxFactory": "createCell",
        "target": "ES5",
        "lib": ["ES2019", "DOM"]
    },
    "include": ["source/**/*"]
}
```
* node version: 12.16.1
* If Windows, are you using WSL or WSL2?: No

**Comment by cspotcode** [comment]
This sounds like an issue that will be fixed by #970 

The TypeScript compiler refuses to compile files with `node_modules` in their path, because it classified these files as "external."  #970 talks about how we can implement our own classification logic.

**Comment by TechQuery** [comment]
@cspotcode But why Windows works? The difference of File System?

**Comment by cspotcode** [comment]
Not sure, though that does seem strange.

Could be an issue with Windows paths using backslashes instead of forward slashes, or an issue with Windows being case-insensitive.

Can you simplify your reproduction to the bare minimum?  I think you can trigger the problem with only 4 files, for example: `tsconfig.json`, `package.json`, `index.ts`, and `node_modules/foo/bar.ts`.

**Comment by TechQuery** [comment]
@cspotcode reproduction created, but GitHub Actions doesn't run...

**Comment by cspotcode** [comment]
Thanks!  The Github Actions is still experimental, I may have made a mistake.  I will take a look.

**Comment by cspotcode** [comment]
I tried running and it succeeds; no errors.  Can you take another look?

I also fixed the Github Actions, so if you merge the latest master into your branch and push again, it should run.

**Comment by TechQuery** [comment]
@cspotcode I don't know why the minimal reproduction is OK...

**Comment by cspotcode** [comment]
@TechQuery I suspect this was working on Windows due to the bug fixed by #997, which was released in [v8.9.0](https://github.com/TypeStrong/ts-node/releases/tag/v8.9.0)

We were not rigorously normalizing Windows paths, so in some codepaths, they had `\` instead of `/`.  Probably, these backslash paths were not being classified as "external" by the compiler, so you were able to require them.

This should be fixed by #970 and looks like the same underlying issue as #876.
